examples/MySQL-client/CMakeLists.txt

```diff
@@ -9,6 +9,10 @@ project(MySQL-client-example)
 hunter_add_package(MySQL-client)
 find_package(MySQL-client REQUIRED)
 
+if(MSVC) #LIBCMT conflicts with MSVCR
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /NODEFAULTLIB:LIBCMT.lib")
```

Comment:
> Therefore anything compiled to use MySQL will have to be /MT as well

Yes, that's the problem. Now users who want to use MySQL should recompile all packages with `/MT` instead of default `/MD`. Which is unmaintainable on practice.

> The only other thing I can think todo is modify MySQL to somehow detect the host applications run-time library setup and set WINDOWS_RUNTIME_MD accordingly

We can't know this a priori. First we install library MySQL-client then we build our project and add MySQL-client paths...

Solution is **to not modify flags** at all, either by removing code `STRING(REPLACE "/MD"  "/MT" "${flag}" "${${flag}}")` or by protecting it with option:

```cmake
# MySQL-client/CMakeLists.txt
option(MYSQL_CLIENT_USE_DEFAULTS "CMake default behaviour" OFF)
if(NOT MYSQL_CLIENT_USE_DEFAULTS)
  # old code to modify compiler flags
  ...
  STRING(REPLACE "/MD"  "/MT" "${flag}" "${${flag}}")
endif()
```

```cmake
# In Hunter: MySQL-client/hunter.cmake
...
hunter_cmake_args(MySQL-client CMAKE_ARGS MYSQL_CLIENT_USE_DEFAULTS=ON)
```